stdlib/nats/src/main/java/org/ballerinalang/nats/nativeimpl/connection/Init.java

```diff
@@ -0,0 +1,118 @@
+        args = {@Argument(name = "config", type = TypeKind.RECORD, structType = "ConnectionConfig",
+                structPackage = "ballerina/nats")},
+        isPublic = true
+)
+public class Init implements NativeCallableUnit {
+
+    private static final String HOST = "host";
+    private static final String PORT = "port";
+    private static final String CLUSTER_ID = "clusterId";
+    private static final String CLIENT_ID = "clientId";
+    private static final String NATS_URL_PREFIX = "nats://";
+    private static final String PROTOCOL_PREFIX = ":";
+    private static final String CONNECTION_TIMEOUT = "connectionTimeout";
+    private static final String MAX_PUB_ACKS_IN_FLIGHT = "maxPubAcksInFlight";
+    private static final String ACK_TIMEOUT = "ackTimeout";
+
+    /**
+     * Initializes NATS streaming connection.
+     *
+     * @param serverConfig Holds values related to establishing a NATS connection.
+     * @return Streaming connection created.
+     * @throws IOException if a failure occurs while establishing a connection.
+     */
+    private StreamingConnection connect(BMap<String, BValue> serverConfig) throws IOException, InterruptedException {
+        String host = ((BString) serverConfig.get(HOST)).value();
+        int port = ((BInteger) serverConfig.get(PORT)).value().intValue();
+        String clusterId = ((BString) serverConfig.get(CLUSTER_ID)).value();
+        String clientId = ((BString) serverConfig.get(CLIENT_ID)).value();
+        long connectionTimeout = ((BInteger) serverConfig.get(CONNECTION_TIMEOUT)).intValue();
+        long mexPubAcksInFlight = ((BInteger) serverConfig.get(MAX_PUB_ACKS_IN_FLIGHT)).intValue();
+        long ackTimeout = ((BInteger) serverConfig.get(ACK_TIMEOUT)).intValue();
+        Options.Builder opts = new Options.Builder().natsUrl(NATS_URL_PREFIX + host + PROTOCOL_PREFIX + port);
+        opts.connectWait(Duration.ofSeconds(connectionTimeout));
+        opts.maxPubAcksInFlight((int) mexPubAcksInFlight);
+        opts.pubAckWait(Duration.ofSeconds(ackTimeout));
+        return NatsStreaming.connect(clusterId, clientId, opts.build());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void execute(Context context, CallableUnitCallback callback) {
+        try {
+            BMap<String, BValue> serverConfig = (BMap<String, BValue>) context.getRefArgument(1);
+            Struct connectorEndpointStruct = BLangConnectorSPIUtil.getConnectorEndpointStruct(context);
+            StreamingConnection connection = connect(serverConfig);
+            connectorEndpointStruct.addNativeData(Constants.NATS_CONNECTION, connection);
+        } catch (IOException e) {
+            Utils.throwBallerinaException("Error occurred while establishing connection", context, e);
+        } catch (InterruptedException ignore) {
+            // ignore
+            Thread.currentThread().interrupt();
+        } catch (Throwable e) {
```

Comment:
Shall we catch Exception instead of Throwable?

Reply:
Same as described in https://github.com/ballerina-platform/ballerina-lang/pull/14669#discussion_r268928258